### PR TITLE
TBD: clean out tapi from repo

### PR DESCRIPTION
**Pull request by cyndyishida**
Now that tbd writer/reader is in llvm, theres no reason to copy/paste the code in swift.

* Removed code and dependecies that swift has to generate tbd files, to instead directly call from LLVM.
* Note this PR is reliant on commits that are in `apple/swift-llvm` in the `upstream-with-swift` branch. Specifically commits
  * 95e7e916d4ac7faac060236e297d2c7bdad32dc3
  * da9c5afc197c0f6a14dabc5f6170fe4a215ff426

To be able to build and test these changes I cherry-picked those commits to the latest `apple/stable` branch within `apple/swift-llvm` and `master-rebranch` within `apple/swift`, which is why my PR is against said branch. 

fix radar 55129229

### CODE REVIEW

**Comment by harlanhaskins** [comment]
Yessssssss, thank you @cyndyishida!!!

**Review by harlanhaskins — APPROVED** [review]

File: lib/TBDGen/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,9 @@
-add_subdirectory(tapi)
-
 add_swift_host_library(swiftTBDGen STATIC
   ${TAPI_SOURCES}
```
Comment (ributzka) [inline lib/TBDGen/CMakeLists.txt]: ${TAPI_SOURCES} is no longer needed or set. I think it should be safe to remove that now too.

**Comment by cyndyishida** [comment]
@swift-ci Please test

**Comment by cyndyishida** [comment]
PR Testing is [disabled](https://ci.swift.org/view/swift-master-rebranch-stable-20190619/job/oss-swift-master-rebranch-pr-test/) for master-rebranch, so I'm going to merge this in, but continue to watch
<https://ci.swift.org/view/swift-master-rebranch-stable-20190619/job/oss-swift-master-rebranch-incremental-RA-osx/> for any regressions this commit could have introduced.

**Comment by shahmishal** [comment]
We are seeing following test failures: 

```
06:02:10 ********************
06:02:10 Failing Tests (14):
06:02:10     Swift(linux-x86_64) :: reproducible-builds/swiftc-emit-tbd.swift
06:02:10     Swift(linux-x86_64) :: TBD/enum.swift
06:02:10     Swift(linux-x86_64) :: TBD/struct.swift
06:02:10     Swift(linux-x86_64) :: TBD/main.swift
06:02:10     Swift(linux-x86_64) :: TBD/dylib-version.swift
06:02:10     Swift(linux-x86_64) :: TBD/protocol.swift
06:02:10     Swift(linux-x86_64) :: TBD/app-extension.swift
06:02:10     Swift(linux-x86_64) :: TBD/global.swift
06:02:10     Swift(linux-x86_64) :: TBD/function.swift
06:02:10     Swift(linux-x86_64) :: TBD/output-path-deduction.swift
06:02:10     Swift(linux-x86_64) :: TBD/class.swift
06:02:10     Swift(linux-x86_64) :: TBD/specialization.swift
06:02:10     Swift(linux-x86_64) :: TBD/opaque_result_type.swift
06:02:10     Swift(linux-x86_64) :: Runtime/backtrace.swift
```
on Ubuntu 18.04. 


Also, you should be able to request PR testing if you see any issue please let me know.

**Pull request merged**